src/mbgl/map/map_impl.cpp

```diff
@@ -181,4 +181,14 @@ void Map::Impl::onStyleImageMissing(const std::string& id, std::function<void()>
     onUpdate();
 }
 
+void Map::Impl::onRemoveUnusedStyleImages(std::vector<std::string> unusedImageIDs) {
+    // If platform does not process onRemoveUnusedStyleImages, by default, remove unused
+    // images from the style.
+    if (!observer.onRemoveUnusedStyleImages(unusedImageIDs)) {
```

Comment:
maybe `onRemoveUnusedStyleImages` shall accept const ref?

Reply:
true, after I moved default handling to core, this can be changed to const ref

Reply:
What is the purpose of passing an array if we are getting a single response? I would have thought that we may get a map between the images we are ok to remove, and those we are not.

Reply:
@fabian-guerra `onRemoveUnusedStyleImages` is a notification, and return value indicates whether this notification was handled by the developer. As for the notification payload, array is more efficient than going over JNI for each single image.

I believe this is rather simple API  yet, provides application developer to control what images may be removed / kept.

Reply:
@alexshalamov what happens if let's say I get a notification with 10 images to remove, and from that array I want to keep one. What would happen in that case?

Reply:
@fabian-guerra as a developer, you would need to remove all but one you need to keep. Passing one-by-one is inefficient, as we already know that there are few images that can be released.